acme/dns_challenge_rfc2136.go

```diff
@@ -99,3 +101,55 @@ func (r *DNSProviderRFC2136) changeRecord(action, fqdn, value string, ttl int) e
 
 	return nil
 }
+
+// findZone determines the zone of a qualifying cert DNSname
+func (r *DNSProviderRFC2136) findZone(fqdn string) (string, error) {
+	// Do we have it cached?
+	if val, ok := r.domain2zone[fqdn]; ok {
+		return val, nil
+	}
+
+	// Query the authorative nameserver for a hopefully non-existing SOA record,
+	// in the authority section of the reply it will have the SOA of the
+	// containing zone. rfc2308 has this to say on the subject:
+	//   Name servers authoritative for a zone MUST include the SOA record of
+	//   the zone in the authority section of the response when reporting an
+	//   NXDOMAIN or indicating that no data (NODATA) of the requested type exists
+	m := new(dns.Msg)
+	m.SetQuestion(fqdn, dns.TypeSOA)
+	m.SetEdns0(4096, false)
+	m.RecursionDesired = true
```

Comment:
Actually, I tried that, didn't work. I got a answer back listing all the root NSs.

Let me try again to see what goes wrong, because I run "dig +norecurse SOA smtp.my.domani" I do get a proper reply, so i'm missing something.